### ISSUE TITLE
For Cycle 64

### ISSUE THREAD

**Issue by alvasw** [bisq-network/compensation#1660]
## Summary

> _Specify the total amount of BSQ you are requesting, along with the USD total and BSQ/USD rate (don't include the brackets!):_

- **BSQ requested**: **`8571.43`** 
- USD requested: 15000 
- BSQ rate: 1.75 USD per BSQ 
- Previous compensation request (if applicable): #1638  

## Contributions delivered

> _Add contributions you have [delivered](https://github.com/bisq-network/proposals/issues/19) and roles you have performed here as new rows in the table below. Role line-items should include an asterisk (*) in the team column._

| Title | Team | USD | Link | Notes |
| --- | --- | ---: | --- | --- |
| bisq: Implement BitcoinCoreSeedListDiscovery method | dev | | https://github.com/bisq-network/bisq/pull/7277 | <ul dir="auto"> <li>Update bitcoin core seed list to upstream version</li> <li>Use IpV4 and onion addresses from it</li> </ul> <p dir="auto">Our users are still protected because we connect to IpV4 addresses<br> through Tor. This improves connectivity because Bitcoin Core creates an<br> ephemeral onion service on each startup.</p> |
| bisq: Migrate Socks5SeedOnionDiscovery to BitcoinCoreSeedListDiscovery | dev | | https://github.com/bisq-network/bisq/pull/7278 | <p dir="auto">Our Socks5SeedOnionDiscovery contains deprecated Tor V2 onion addresses<br> that are not supported anymore. BitcoinCoreSeedListDiscovery contains<br> Tor V3 addresses that are shipped with Bitcoin Core.</p> |
| bisq: DnsLookupTor: Close socket after usage | dev | | https://github.com/bisq-network/bisq/pull/7279 |  |
| bisq: Socks5MultiDiscovery: Don't create useless array | dev | | https://github.com/bisq-network/bisq/pull/7280 | <p dir="auto">We create an array in the getPeers method and throw it away without<br> using it.</p> |
| bisq: Socks5MultiDiscovery: Shutdown peer discovery methods | dev | | https://github.com/bisq-network/bisq/pull/7281 |  |
| bisq: Add bitcoind submodule | dev | | https://github.com/bisq-network/bisq/pull/7282 | <p dir="auto">We use the same bitcoind regtest module in Bisq 1 and 2. To avoid<br> duplicate code I moved the bitcoind module to its own repository.</p> |
| bisq: core: Implement transaction offset signing | dev | | https://github.com/bisq-network/bisq/pull/7283 | <p dir="auto">The BisqTransactionSigner signs the inputs of a transaction after the<br> given offset using the LocalOffsetTransactionSigner. The<br> LocalOffsetTransactionSigner is identical to BitcoinJ's<br> LocalTransactionSigner. The only difference is that the<br> LocalOffsetTransactionSigner accepts an offset in its constructor. All<br> inputs below this offset are skipped during signing.</p> <p dir="auto">This is needed for Bisq's BSQ implementation because the mining fees of<br> BSQ transactions come from Bisq's BTC wallet. BitcoinJ's<br> LocalTransactionSigner expects all inputs to be part of the same wallet.</p> |
| bisq: BtcWalletV2: Implement addMiningFeesToBsqTx | dev | | https://github.com/bisq-network/bisq/pull/7284 | <p dir="auto">The addMiningFeesToBsqTx method adds BTC mining fees to a given BSQ<br> transaction and tx fee amount.</p> |
| bisq: BsqWalletV2: Implement send BSQ | dev | | https://github.com/bisq-network/bisq/pull/7285 | <p dir="auto">The sendBsq method creates a BSQ transaction, signs the BSQ inputs, and<br> adds the requires BTC mining fees to the transaction.</p> |
| bisq: PR Review bisq-network/bisq#7286 | dev | | https://github.com/bisq-network/bisq/pull/7286 | <p dir="auto">utACK</p> |
| bisq: core: Implement WalletFactory | dev | | https://github.com/bisq-network/bisq/pull/7287 | <p dir="auto">The WalletFactory can create BTC and BSQ wallets.</p> |
| bisq: gradle: Fix verification-metadata.xml formatting | dev | | https://github.com/bisq-network/bisq/pull/7288 |  |
| bisq: gradle: Create bisq.java-integration-tests plugin | dev | | https://github.com/bisq-network/bisq/pull/7289 | <p dir="auto">The bisq.java-integration-tests plugin creates an integrationTest source<br> set that depends on JUnit. The source set inherits the main source set's<br> compile classpath, runtime classpath, and the implementation's source<br> set configuration and runtimeOnly configuration.</p> |
| bisq: core: Implement RegtestWalletAppKit | dev | | https://github.com/bisq-network/bisq/pull/7290 | <p dir="auto">The RegtestWalletAppKit sets up BitcoinJ and loads the given wallets.</p> |
| bisq: bitcoind: Target commit 6737755 | dev | | https://github.com/bisq-network/bisq/pull/7291 |  |
| bisq: core: Add dependency to bitcoind:regtest (integration source set) | dev | | https://github.com/bisq-network/bisq/pull/7292 |  |
| bisq: core: Implement sendBsqTest | dev | | https://github.com/bisq-network/bisq/pull/7293 | <p dir="auto">First, the sendBsq test creates one BTC and two BSQ wallets. Afterward,<br> it funds the BTC and one BSQ wallet with 1 BTC. Next, the funded BSQ<br> wallet sends 100 BSQ to the second BSQ wallet.</p> |
| bisq: sendBsq: Throw InsufficientMoneyException when BSQ balance too low | dev | | https://github.com/bisq-network/bisq/pull/7294 | <p dir="auto">We should throw an InsufficientMoneyException when the CoinSelection<br> gathered amount is lower than the amount the user wants to send.</p> |
| bisq: BsqWalletV2Test: Add invalid and dust send tests | dev | | https://github.com/bisq-network/bisq/pull/7295 | <p dir="auto">The tests try to send 0 BSQ and the current dust amount in BSQ.</p> |
| bisq2: Tor: Ship all required shared libraries with Bisq | dev | | https://github.com/bisq-network/bisq2/pull/2938 | <p dir="auto">Tor failed to start for some users because their operating system didn't ship Tor's dependencies. This change instructs the package manager to install tor and all its required during the Bisq installation. Furthermore, we don't need to ship the Tor binary and its libraries on Linux anymore leading to package size savings.</p> <p dir="auto">Changes:</p> <ul dir="auto"> <li><a href="https://github.com/bisq-network/bisq2/commit/d378a423509a73b9fb8340f70360c23ed4cca421">packaging: Add tor package to required Linux dependencies</a></li> <li><a href="https://github.com/bisq-network/bisq2/commit/181f970d5400a1e856b8a96e90b1841d75028daf">TorService: Remove duplicate call to installTorIfNotUpToDate()</a></li> <li><a href="https://github.com/bisq-network/bisq2/commit/4c791be7ab95cdcacd474302cf8ee52297252684">tor: Use system tor (installed by package manager)</a></li> <li><a href="https://github.com/bisq-network/bisq2/commit/246535de503a734b526ea573cf35cf9dd7c5ec02">build-logic: Don't bundle Tor on Linux</a></li> </ul> <p dir="auto"><span class="issue-keyword tooltipped tooltipped-se" aria-label="This pull request closes issue #2906.">Fixes</span> <a class="issue-link js-issue-link" data-error-text="Failed to load title" data-id="2568290748" data-permission-text="Title is private" data-url="https://github.com/bisq-network/bisq2/issues/2906" data-hovercard-type="issue" data-hovercard-url="/bisq-network/bisq2/issues/2906/hovercard" href="https://github.com/bisq-network/bisq2/issues/2906">#2906</a></p> |
| bisq2: Implement MigrationService | dev | | https://github.com/bisq-network/bisq2/pull/2939 | <p dir="auto">The MigrationService allows us to perform necessary migrations when upgrading to a new Bisq version. So far, the only option is to try to perform the migrations on each run. The MigrationService runs the necessary migrations if the data directory version is below the app version.</p> <p dir="auto">I tested this PR on:</p> <ul dir="auto"> <li>an instance with a fresh data directory</li> <li>an instance with a data directory of v2.1.1 (applies migration)</li> <li>an instance with an already migrated data directory (v2.1.2)</li> </ul> <p dir="auto">This feature is needed to remove redundant Tor files after merging PR <a class="issue-link js-issue-link" data-error-text="Failed to load title" data-id="2599702700" data-permission-text="Title is private" data-url="https://github.com/bisq-network/bisq2/issues/2938" data-hovercard-type="pull_request" data-hovercard-url="/bisq-network/bisq2/pull/2938/hovercard" href="https://github.com/bisq-network/bisq2/pull/2938">#2938</a>. A follow-up PR will contain the required changes.</p> <p dir="auto">Changes:</p> <ul dir="auto"> <li><a href="https://github.com/bisq-network/bisq2/commit/31167e00930479788145dec50c372c2c10881683">Implement MigrationService</a></li> <li><a href="https://github.com/bisq-network/bisq2/commit/27f1a6ad1d9627c92a25ce7488254b41a414b11d">Version: Throw InvalidVersionException for invalid versions</a></li> <li><a href="https://github.com/bisq-network/bisq2/commit/9d8187dffd875bcc6a9a2129ca7d20f20cbcda6a">Test MigrationService data directory version detection</a> <ul dir="auto"> <li>Test with data directory before MigrationService was introduced</li> <li>Test with invalid data directory version</li> <li>Test with valid data directory version</li> </ul> </li> <li><a href="https://github.com/bisq-network/bisq2/commit/c216ccab54f93fe74b46435d4e44488564abf234">Implement migration for v2.1.2</a> <ul dir="auto"> <li>This migration writes the current version number to the data directory. The remaining migrations depend on another open PR and will follow soon.</li> </ul> </li> <li><a href="https://github.com/bisq-network/bisq2/commit/ff2dca8a70cd6335d2ffd95b24b4fab46e7e38d3">ApplicationService: Apply migrations before starting services</a></li> </ul> |
| bisq2: migration: Remove tor files from prior version on Linux | dev | | https://github.com/bisq-network/bisq2/pull/2941 | <p dir="auto">PR <a class="issue-link js-issue-link" data-error-text="Failed to load title" data-id="2599702700" data-permission-text="Title is private" data-url="https://github.com/bisq-network/bisq2/issues/2938" data-hovercard-type="pull_request" data-hovercard-url="/bisq-network/bisq2/pull/2938/hovercard" href="https://github.com/bisq-network/bisq2/pull/2938">#2938</a> unbundles Tor from Bisq on Linux and switches to the OSes Tor binary. If a user updates from a prior version their data directory will contain the old extracted Tor binaries. The migration in this change removes the obsolete files.</p> |
| bisq2: migration: Write new data directory version if all migrations succeed | dev | | https://github.com/bisq-network/bisq2/pull/2946 | <p dir="auto">Migrations are applied if the app version number is higher than the data directory version number. Currently, the only existing migration writes the new version number to the data directory after applying its migration. Moving the data directory version write to the Migrator makes the code future-proof when we have multiple migrations.</p> |
| bisq2: migration: Pass migrations to Migrator in constructor | dev | | https://github.com/bisq-network/bisq2/pull/2974 | <p dir="auto">The migrations are hard-coded in the Migrator making it impossible to add test where some migrations fail.</p> |
| bisq2: migration: Test migration failures | dev | | https://github.com/bisq-network/bisq2/pull/2975 |  |
| bisq2: migration: Don't keep MigrationService in memory | dev | | https://github.com/bisq-network/bisq2/pull/2976 | <p dir="auto">The MigrationService isn't needed anymore after the data directory has<br> been migrated. This change removes the ApplicationService's reference to<br> the MigrationService after it has run.</p> |
| bisq2: Fix CI (macOS 12 deprecated) | dev | | https://github.com/bisq-network/bisq2/pull/2977 | <p dir="auto">The CI tasks are failing because macOS 12 isn't supported anymore. See: <a class="issue-link js-issue-link" data-error-text="Failed to load title" data-id="2560034362" data-permission-text="Title is private" data-url="https://github.com/actions/runner-images/issues/10721" data-hovercard-type="issue" data-hovercard-url="/actions/runner-images/issues/10721/hovercard" href="https://github.com/actions/runner-images/issues/10721">actions/runner-images#10721</a></p> |
| bisq2: Add wallets/bitcoind submodule | dev | | https://github.com/bisq-network/bisq2/pull/2984 | <p dir="auto">We use the same bitcoind regtest module in Bisq 1 and 2. To avoid<br> duplicate code I moved the bitcoind module to its own repository.</p> |
| bitcoind: Add .gitignore from root | dev | | https://github.com/bisq-network/bitcoind/commit/5571e273d30ba44afc374525aab24687f36427d5 | |
| bitcoind: gradle-tasks: Replace .lowercase(Locale.US) by .toLowerCase() | dev | | https://github.com/bisq-network/bitcoind/commit/b657101b7ad73caa8d6e4046557c96d66f391028 | |
| bitcoind: gradle: Set build-logic's name to bitcoind-build-logic | dev | | https://github.com/bisq-network/bitcoind/commit/ba4bc2530188346e1281b09225f265c6e7434a0c | |
| bitcoind: gradle: Set JDK version from 22 to 11 | dev | | https://github.com/bisq-network/bitcoind/commit/6737755da1e2c146459a55ddac942101fade86a7 | |
| | dev | 15000 | | Total for items above.

**Comment by alvasw** [comment]
a03544ea962acafaed5c3c8019bd7c6ef1917b0a83f0a058cb73ec3cb283be21

**Comment by MwithM** [comment]
Proposal accepted

**Comment by MwithM** [comment]
Was accepted.
